### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by MBueschelberger**
* add unit conversion/fetching utility
* add unit conversion/fetching to dataframe columns
* add unit conversion/fetching to custom properties, generally update the pydantic model
* add semantics module for exchangable semantic queries via config
* add option to directly source .env file via python dotenv
* add unit tests
* add updating of avatar
* fix problems in commiting linked kitems 
* add running apps on demand
* add interaction with app configs
* add docs
* add logging levels
* add avatar interaction
* add functionality to add attachment content from memory
* remove side effects of KProperties when added, updated, deleted, retrieved.
* update pydantic models for linked kitems

### CODE REVIEW

File: dsms/knowledge/utils.py [hunk 1]
```diff
@@ -2,48 +2,162 @@
...
-            dtype = float
-        elif dtype == "bool":
-            dtype = bool
-        elif dtype == "object":
-            dtype = dict
-        else:
-            raise TypeError(f"Invalid `type={dtype}`")
-        fields[key] = (dtype, ... if not default else default)
-    return create_model(title, **fields)
+    if isinstance(value, dict):
```
Comment (yoavnash) [inline dsms/knowledge/utils.py:48]: I had to comment out this block due to the latest changes done related to the integration of form2rdf into DSMS

File: examples/unit_conversion.ipynb [hunk 2]
```diff
@@ -0,0 +1,1265 @@
...
+       " ...]"
+      ]
+     },
+     "execution_count": 19,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "uuid = \"8d6b034e-efb4-427a-a0d3-666e62762cba\"\n",
```
Comment (yoavnash) [inline examples/unit_conversion.ipynb:1234]: I'm not sure if it's a good idea to assume this kitem exists. This part failed for me when running it on my local instance. Maybe check first if it exists and if not, create it?

**Comment by MBueschelberger** [comment]
@nash, this is ready for review, only #21 needs to be done but that's a minor change.

File: examples/unit_conversion.ipynb (hunk 2 above)
Comment (MBueschelberger) [reply examples/unit_conversion.ipynb]: Notebook has been deleted.

**Pull request merged**